cmd/juju/status/output_tabular.go

```diff
@@ -165,6 +165,24 @@ func printApplications(tw *ansiterm.TabWriter, fs formattedStatus) {
 			if len(parts) > 1 && len(version) > maxVersionWidth {
 				version = strings.Join(parts[1:], "/")
 			}
+			// Charms deployed from the rocks or jujucharm repos have the
+			// image path set to <repo-url>/<charm_name>/<resource_name>. Charm name
+			// is somewhat redundant and takes up value space. So we'll replace
+			// with "ch:" to indicate the named resource comes from the store repo.
+			parts = strings.Split(version, "/")
+			if len(parts) > 1 && (len(version) > maxVersionWidth || parts[0] == app.Charm) {
+				prefix := ""
+				switch app.CharmOrigin {
+				case "charmhub":
+					prefix = "ch"
+				case "charmstore":
+					prefix = "cs"
+				}
```

Comment:
So what do we get if the CharmOrigin isn't either of these? eg: '--resource oci-image=jameinel/mysql'

It seems like this would also truncate those to just 'mysql'. Which would confuse "oci-image=mysql:latest' with "jameinel/mysql:latest".

I wonder if we can have a better way to only do the path stripping if it is a resource, and have a clear way to say it is a resource. 

`resource:oci-image@deadbee`

is probably a bit long, but does convey it clearly. And I do know that Mark has specifically stated he doesn't like to see 'ch:' anywhere.

`r:` feels like it is unclear what you mean by r. `res:oci-image@deadbee` does make efficient use of the space, but does still use a name that isn't defined.

Reply:
It's difficult to use a very short text string to properly display something meaningful with all the possible permutations. There's  official upstream images from dicker.io or quay.io etc; there's images from someone's namespace on a repo; there's images from a canonical repo which use the charm declared resource name and which obfuscates the original image source. Anything other than the image name and sha very quickly uses the available width.
I think "res:" is probably the best compromise.